pkg/kapp/diff/explicit_versioned_ref.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package diff
+
+import (
+	"fmt"
+
+	ctlres "github.com/k14s/kapp/pkg/kapp/resources"
+	"gopkg.in/yaml.v2"
+)
+
+type ExplicitVersionedRef struct {
+	AnnotationKey string
+	Annotation    string
+}
+
+const (
+	explicitReferenceKey       = "kapp.k14s.io/versioned-explicit-ref"
+	explicitReferenceKeyPrefix = "kapp.k14s.io/versioned-explicit-ref."
+)
+
+func NewExplicitVersionedRef(annotationKey string, annotation string) *ExplicitVersionedRef {
+	return &ExplicitVersionedRef{annotationKey, annotation}
+}
+
+func (e *ExplicitVersionedRef) AsObjectRef() (map[string]interface{}, error) {
+	var objectRef map[string]interface{}
+	err := yaml.Unmarshal([]byte(e.Annotation), &objectRef)
+	if err != nil {
+		return nil, fmt.Errorf("Explicit reference error: could not create object ref for explicit versioned reference")
+	}
+
+	if objectRef["apiVersion"] == "" || objectRef["kind"] == "" || objectRef["name"] == "" {
+		return nil, fmt.Errorf("Explicit versioned reference error: apiVersion, kind and name are required values in an explicit versioned reference")
+	}
```

Comment:
`fmt.Errorf("Expected versioned explicit reference to specify non-empty apiVersion, kind and name keys")`

Reply:
`fmt.Errorf("Expected versioned  reference to specify non-empty apiVersion, kind and name keys")` maybe? Typo?